light/beacon/sync_committee_head.go

```diff
@@ -0,0 +1,304 @@
+			log.Warn("Old signed head received", "age", age)
+		}
+		if !sigOk {
+			peer.WrongReply("invalid header signature")
+			continue
+		}
+		hash := head.Header.Hash()
+		if h := s.acceptedList.getHead(hash); h != nil {
+			h.receivedFrom[peer] = struct{}{}
+			if signerCount > h.signerCount {
+				h.head = head
+				h.signerCount = signerCount
+				h.sentTo = nil
+				s.acceptedList.updateHead(h)
+				if h.processed {
+					s.processedList.updateHead(h)
+					broadcast = true
+				}
+			}
+		} else {
+			var processed bool
+			for _, p := range s.lastProcessed {
+				if p == hash {
+					processed = true
+					break
+				}
+			}
+			h := &headInfo{
+				head:         head,
+				hash:         hash,
+				sentTo:       make(map[sctClient]struct{}),
+				receivedFrom: map[sctServer]struct{}{peer: struct{}{}},
+				processed:    processed,
+			}
+			s.acceptedList.updateHead(h)
+			if h.processed {
+				s.processedList.updateHead(h)
+				broadcast = true
+			}
+		}
+	}
+	if broadcast {
+		s.broadcastHeads()
+	}
+	if len(s.acceptedList.list) > 0 && oldHeadHash != s.acceptedList.list[0].hash {
+		head := s.acceptedList.list[0].head.Header
+		for _, subFn := range s.headSubs {
+			subFn(head)
+		}
+	}
+}
+
+// verifySignature returns true if the given signed head has a valid signature according to the local
+// committee chain. The caller should ensure that the committees advertised by the same source where
+// the signed head came from are synced before verifying the signature.
+// The age of the header is also returned (the time elapsed since the beginning of the given slot,
+// according to the local system clock). If enforceTime is true then negative age (future) headers
+// are rejected.
+func (s *SyncCommitteeTracker) verifySignature(head SignedHead) (bool, time.Duration) {
+	slotTime := int64(time.Second) * int64(s.genesisTime+uint64(head.Header.Slot)*12)
+	age := time.Duration(s.unixNano() - slotTime)
+	if s.enforceTime && age < 0 {
+		return false, age
+	}
+	committee := s.getSyncCommittee(uint64(head.Header.Slot+1) >> 13) // signed with the next slot's committee
+	if committee == nil {
+		return false, age
+	}
+	return s.sigVerifier.verifySignature(committee, s.forks.signingRoot(head.Header), head.BitMask, head.Signature), age
+}
+
+// SubscribeToNewHeads subscribes the given callback function to head beacon headers with a verified valid sync committee signature.
+func (s *SyncCommitteeTracker) SubscribeToNewHeads(subFn func(Header)) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.headSubs = append(s.headSubs, subFn)
+}
+
+// AllowBroadcast should be called when given head can be broadcast to other peers. Depending on the setup further data retrieval and/or
+// processing might be needed for a new head before it is propagated further.
+// Note that a signed head is broadcast when both addSignedHeads and AllowBroadcast has been called for the given head (in any order).
+func (s *SyncCommitteeTracker) AllowBroadcast(hash common.Hash) {
```

Comment:
The chain head broadcast mechanism is not used in this PR. I know it might be used in the following PR, but I will vote to not include these logics in this PR to minimize the PR size.

Reply:
Indeed, I removed most of the unused code but not this one. Done now.